CMSgov/cmcs-eregulations
# EREGCSC-1105 SPA Prototype

Resolves [EREGCSC-1105](https://jiraent.cms.gov/browse/EREGCSC-1105)

- Adds `prototype` folder to `/regulations/static`.
- Used `vue-cli` tooling create a single page app (SPA) prototype project scaffolding and add plugins.
- Enables live/hot reloading on file change.
- Dockerizes the SPA so it runs a local dev server in a container so it "works on my machine" for everyone.
- Adds Vuetify component framework to support rapid prototyping.
- Allows for external dependencies to be easily added to project.  Tested with Lodash.
- Adds Vue Router for client side routing
- Adds Jest for unit testing

**Steps to manually verify this change:**

1. run `make prototype` from repository root
2. visit `localhost:8081`
3. Make change to component in `/regulations/static/prototype/src/components/HelloWorld.vue` to observe live reloading
4. Visit `localhost:8081/about` to observe routing
5. run `make prototype:clean` from repository root to tear down docker container
6. run `npm run test:unit` from `/regulations/static/prototype` to see passing unit test
7. edit HelloWorld spec at `/regulations/static/prototype/src/components/HelloWorld.spec.js` to make unit test fail
